dxp/layout_rendering.md

```diff
@@ -0,0 +1,173 @@
+	_publicPaths.add(_PATH_PORTAL_LAYOUT);
+	_publicPaths.add(_PATH_PORTAL_LICENSE);
+	_publicPaths.add(_PATH_PORTAL_LOGIN);
+	_publicPaths.add(_PATH_PORTAL_RENDER_PORTLET);
+	_publicPaths.add(_PATH_PORTAL_TCK);
+	_publicPaths.add(_PATH_PORTAL_UPDATE_LANGUAGE);
+	_publicPaths.add(_PATH_PORTAL_UPDATE_PASSWORD);
+	_publicPaths.add(_PATH_PORTAL_VERIFY_EMAIL_ADDRESS);
+	_publicPaths.add(PropsValues.AUTH_LOGIN_DISABLED_PATH);
+```
+
+`MainServlet` delegates to `PortalRequestProcessor.process()` the handling of every `GET` and `POST` request.
+
+In addition it may redirect to the _Setup Wizard_ or the default _Error Page_.
+
+It contains a map named `_definitions` which is loaded from the servlet context invoking:
+
+```java
+	servletContext.getAttribute(TilesUtil.DEFINITIONS)
+```
+
+and maps Struts' logical forwards (for example: `portal.layout`) to paths inside DXP (for example: `/common/themes/portal.jsp`).
+
+## `Action`
+
+It is an implementation that lives inside `portal-impl` and emulates the old `StrutsAction`. It was created to remove Struts from DXP while still retaining the old structure of Java classes and prevent migrating them to portlets, servlets, or anything similar.
+
+Currently we have the following `Action`s in `portal-impl`:
+
+-   `portal-impl/src/com/liferay/portal/struts/model/ActionMapping.java`
+-   `portal-impl/src/com/liferay/portal/struts/model/ModuleConfig.java`
+-   `portal-impl/src/com/liferay/portal/action/VerifyEmailAddressAction.java`
+-   `portal-impl/src/com/liferay/portal/action/UpdateTermsOfUseAction.java`
+-   `portal-impl/src/com/liferay/portal/action/LoginAction.java`
+-   `portal-impl/src/com/liferay/portal/action/RESTProxyAction.java`
+-   `portal-impl/src/com/liferay/portal/action/UpdatePasswordAction.java`
+-   `portal-impl/src/com/liferay/portal/action/UpdateLayoutAction.java`
+-   `portal-impl/src/com/liferay/portal/action/UpdateLicenseAction.java`
+-   `portal-impl/src/com/liferay/portal/action/SetupWizardAction.java`
+-   `portal-impl/src/com/liferay/portal/action/SitemapAction.java`
+-   `portal-impl/src/com/liferay/portal/action/SessionClickAction.java`
+-   `portal-impl/src/com/liferay/portal/action/UpdateReminderQueryAction.java`
+-   `portal-impl/src/com/liferay/portal/action/RenderPortletAction.java`
+-   `portal-impl/src/com/liferay/portal/action/UpdateLanguageAction.java`
+-   `portal-impl/src/com/liferay/portal/action/LogoutAction.java`
+-   `portal-impl/src/com/liferay/portal/action/TestAction.java`
+-   `portal-impl/src/com/liferay/portal/action/TestRedirectAction.java`
+-   `portal-impl/src/com/liferay/portal/action/SessionTreeJSClickAction.java`
+-   `portal-impl/src/com/liferay/portal/action/RobotsAction.java`
+-   `portal-impl/src/com/liferay/portal/action/UpdateEmailAddressAction.java`
+-   `portal-impl/src/com/liferay/portal/action/LayoutAction.java`
+
+Note that, because this is legacy code, we should expect no more `Action`s to appear in DXP's codebase.
+
+## `StrutsAction`
```

Comment:
Similar to the other actions, what's the relation to rendering a layout?

Reply:
Nothing, maybe we should move this to a different document.

Reply:
I have left the references here, but explained they are not true part of the layout rendering process, in other words: I only mention them in the document to add random noise.